### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by shm04**
### In this pull request I add the following changes: 

- Create a class Classroom with the following:
  - @label instance variable, should be initialized in the constructor.
  - Setter and getter for @label (remember about attr_accessor).
- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
  - Create the has-many side (a classroom has many students).
  - Create the belongs-to side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class Book with the following:
  - @title and @author instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about attr_accessor).
- Create a class Rental with the following:
  - @date instance variable, should be initialized in the constructor.
  - Setter and getter for @date (remember about attr_accessor).
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
   - Create the has-many side of Book and Rental (a book has many rentals).
  - Create the belongs-to side of Rental and Book (a rental belongs to a book).
  - Create the has-many side of Person and Rental (a person has many rentals).
  - Create the belongs-to side of Rental and Person (a rental belongs to a person).
  - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+end
+
+class Student
+  attr_accessor :name, :classroom
+
+  def initialize(name)
+    @name = name
+    @classroom = nil
+  end
+end
```
Comment (rotimiazeez) [inline classroom.rb]: Please ensure your student class has a method that serves as the 'belongs-to' side (a student belongs to a classroom)👍🏽 
```ruby
 def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: classroom.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (rotimiazeez) [inline classroom.rb]: Please ensure your Book class has a method for adding rentals 👍🏽 
```ruby
def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    Rental.new(date, self, person)
  end
```

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+class Person
+  attr_accessor :name, :rentals
+
+  def initialize(name)
+    @name = name
+    @rentals = []
+  end
+end
```
Comment (rotimiazeez) [inline classroom.rb]: Please ensure your Person class has a method for adding rentals 👍🏽
```ruby
def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @shm04 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- Good job so far with creating the associations 👌🏽
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Descriptive README 📖
You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
### It will be nice to have all your classes in different files as it makes your code well-structured and organized 😉
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

**Review by Munsa1 — APPROVED** [review]
Hi @shm04 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Made Requested Changes By Previous Reviewer :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**